plugins/modules/packaging/os/yum_versionlock.py

```diff
@@ -0,0 +1,149 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2018, Florian Paul Hoberg <florian.hoberg@credativ.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: yum_versionlock
+version_added: 2.0.0
+short_description: Locks / Unlocks an installed package(s) from being updated by yum package manager.
+description:
+     - This module adds installed packages to yum versionlock to prevent the package from being updated.
+options:
+  name:
+    description:
+      - Package name or a list of packages.
+    type: list
+    required: True
+    elements: str
+  state:
+    description:
+    - If state is C(present) or C(locked), package(s) will be added to yum versionlock list.
+    - If state is C(absent) or C(unlocked), package(s) will be removed from yum versionlock list.
+    choices: [ 'absent', 'locked', 'present', 'unlocked' ]
+    type: str
+    default: present
+notes:
+    - Requires yum-plugin-versionlock package on the remote node.
+    - Supports C(check_mode).
+requirements:
+- yum
+- yum-versionlock
+author:
+    - Florian Paul Hoberg (@florianpaulhoberg)
+'''
+
+EXAMPLES = r'''
+- name: Prevent Apache / httpd from being updated
+  community.general.yum_versionlock:
+    state: present
+    name: httpd
+
+- name: Prevent multiple packages from being updated
+  community.general.yum_versionlock:
+    state: present
+    name:
+    - httpd
+    - nginx
+    - haproxy
+    - curl
+
+- name: Unlock Apache / httpd to be updated again
+  community.general.yum_versionlock:
+    state: absent
+    package: httpd
+'''
+
+RETURN = r'''
+packages:
+    description: A list of package(s) in versionlock list.
+    returned: everytime
+    type: list
+    sample: [ 'httpd' ]
+state:
+    description: State of package(s).
+    returned: everytime
+    type: str
+    sample: present
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+
+class YumVersionLock:
+    def __init__(self, module):
+        self.module = module
+        self.params = module.params
+        self.yum_bin = module.get_bin_path('yum', required=True)
+
+    def get_versionlock_packages(self):
+        """ Get an overview of all packages on yum versionlock """
+        rc, out, err = self.module.run_command("%s versionlock list" % self.yum_bin)
```

Comment:
Please always use lists for `run_command` arguments.
```suggestion
        rc, out, err = self.module.run_command([self.yum_bin, "versionlock", "list"])
```